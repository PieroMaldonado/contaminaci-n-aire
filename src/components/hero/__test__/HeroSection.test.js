/* eslint-disable comma-dangle */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import store from '../../../redux/configureStore';
import HeroSection from '../HeroSection';

afterEach(cleanup);
describe('Hero Component', () => {
  test('Render Hero', () => {
    render(
      <Router>
        <Provider store={store}>
          <HeroSection />
        </Provider>
      </Router>
    );

    const buttonElement = screen.getByTestId('button');
    const heroElement = screen.getByTestId('header_name');

    expect(heroElement).toBeInTheDocument();
    expect(buttonElement).toBeInTheDocument();
  });

  test('Render Snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <HeroSection />
          </Provider>
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
