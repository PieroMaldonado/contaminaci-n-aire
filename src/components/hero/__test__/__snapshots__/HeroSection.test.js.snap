// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component Render Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="split_section"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      className="left"
    >
      <img
        alt="map"
        className="image"
        src="https://raw.githubusercontent.com/Ginohmk/worldMaps/main/maps// /vector.svg"
      />
    </div>
    <div
      className="right"
    >
      <p
        data-testid="header_name"
      >
         
      </p>
      <span>
         
         
         
        countries
         
      </span>
      <form
        className="form_style"
      >
        <select
          name="selected"
          onChange={[Function]}
        >
          <option>
             
             
            *
             
          </option>
          <option
            value="Asia"
          >
             Asia
          </option>
          <option
            value="Oceania"
          >
            Oceania 
          </option>
          <option
            value="Europe"
          >
             Europe 
          </option>
          <option
            value="Americas"
          >
             Americas
          </option>
          <option
            value="Africa"
          >
             Africa
          </option>
        </select>
        <a
          className=""
          href="/country/Europe"
          onClick={[Function]}
        >
          <button
            data-testid="button"
            type="button"
          >
             
            Search Region
          </button>
        </a>
      </form>
    </div>
  </div>
</div>
`;
